tiledb/tests/test_hypothesis.py

```diff
@@ -20,7 +20,7 @@
 def test_bytes_npdf(checked_path, mode, data):
     start = time.time()
 
-    uri = "mem://" + checked_path.path()
+    uri = checked_path.path()
```

Comment:
I think we might have an issue here with using the same context consistently for array operations, because `mem://` state is attached to the context.

Reply:
> I think we need to dig in to this one more, because it doesn't exactly have multiple dtypes -- the dtype is a record type.

The `DataType` class supports that feature, and I used it.

> I think we might have an issue here with using the same context consistently for array operations, because `mem://` state is attached to the context.

No changes related to this test anymore.